python/pyspark/sql/connect/session.py

```diff
@@ -474,35 +475,40 @@ def createDataFrame(
 
         _table: Optional[pa.Table] = None
 
-        if isinstance(data, pd.DataFrame):
+        if isinstance(data, pd.DataFrame) or isinstance(data, pa.Table):
             # Logic was borrowed from `_create_from_pandas_with_arrow` in
             # `pyspark.sql.pandas.conversion.py`. Should ideally deduplicate the logics.
 
             # If no schema supplied by user then get the names of columns only
             if schema is None:
```

Comment:
quick comment, I wonder if we can flatten this if-else via:

```python
if schema is None and isinstance(data, pd.DataFrame):
    ...
elif schema is None and  isinstance(data, pa.Table)
    ...
```

Then less conflicts to other branches if we land some fixes in the future, easier to backport/revert, etc.

Reply:
I flattened the conditionals. Please take a look: https://github.com/apache/spark/pull/46529/commits/795d01a1fd96fcbe80432d74be50382d350edf57?diff=split&w=0